src/Middleware/RetryMiddleware.php

```diff
@@ -74,121 +76,91 @@ public function __construct(
     public function __invoke(Call $call, array $options)
     {
         $nextHandler = $this->nextHandler;
+        $retrySettings = $this->retrier->getRetrySettings();
 
         if (!isset($options['timeoutMillis'])) {
             // default to "noRetriesRpcTimeoutMillis" when retries are disabled, otherwise use "initialRpcTimeoutMillis"
-            if (!$this->retrySettings->retriesEnabled() && $this->retrySettings->getNoRetriesRpcTimeoutMillis() > 0) {
-                $options['timeoutMillis'] = $this->retrySettings->getNoRetriesRpcTimeoutMillis();
-            } elseif ($this->retrySettings->getInitialRpcTimeoutMillis() > 0) {
-                $options['timeoutMillis'] = $this->retrySettings->getInitialRpcTimeoutMillis();
+            if (!$retrySettings->retriesEnabled() && $retrySettings->getNoRetriesRpcTimeoutMillis() > 0) {
+                $options['timeoutMillis'] = $retrySettings->getNoRetriesRpcTimeoutMillis();
+            } elseif ($retrySettings->getInitialRpcTimeoutMillis() > 0) {
+                $options['timeoutMillis'] = $retrySettings->getInitialRpcTimeoutMillis();
```

Comment:
After discussing with @saranshdhingra, I think having a `parameterUpdateFunction` as part of either `Retrier` or or `RetrySettings` is the way to handle this - we need to update this on every call, but only for calls made through `RetryMiddleware`, so it should be generic. 

I have a WIP here: https://github.com/googleapis/gax-php/commit/6a940361119b0b5bcd682a729f11d4be50e3d3ba